src/material_library.cpp

```diff
@@ -170,7 +170,8 @@ std::string pyne::MaterialLibrary::ensure_material_name(
     }
   } else {
     if (mat_number == -1) {
-      mat_number = keylist.size();  // set a temp mat_number to form mat name
+      mat_number =
+          *max_element(mat_number_set.begin(), mat_number_set.end()) + 1;
```

Comment:
There may be a more efficient way to do this: https://stackoverflow.com/questions/1342045/how-do-i-find-the-largest-int-in-a-stdsetint

Reply:
the problem with that is we will have to deal with the case were the library is empty.